omod/src/main/java/org/openmrs/module/smartonfhir/web/controller/SmartOnFhirRestController.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.module.smartonfhir.web.controller;
+
+import javax.crypto.spec.SecretKeySpec;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import java.io.IOException;
+import java.net.URLDecoder;
+import java.net.URLEncoder;
+import java.nio.charset.StandardCharsets;
+
+import org.keycloak.crypto.Algorithm;
+import org.keycloak.crypto.JavaAlgorithm;
+import org.keycloak.crypto.KeyWrapper;
+import org.keycloak.crypto.MacSignatureSignerContext;
+import org.keycloak.crypto.SignatureSignerContext;
+import org.keycloak.jose.jws.JWSBuilder;
+import org.keycloak.representations.JsonWebToken;
+import org.openmrs.module.smartonfhir.util.SmartSecretKeyHolder;
+import org.openmrs.module.webservices.rest.web.RestConstants;
+import org.openmrs.module.webservices.rest.web.v1_0.controller.BaseRestController;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+
+@Controller
+@RequestMapping("/rest/" + RestConstants.VERSION_1 + SmartOnFhirRestController.SMARTONFHIR_NAMESPACE)
+public class SmartOnFhirRestController extends BaseRestController {
+	
+	public static final String SMARTONFHIR_NAMESPACE = "/smartonfhir";
+	
+	@RequestMapping(value = "/accessConformation", method = RequestMethod.GET)
+	public void accessConformation(HttpServletRequest request, HttpServletResponse response) throws IOException {
+		
+		String token = request.getParameter("token");
+		
+		if (token == null) {
+			response.sendError(HttpServletResponse.SC_BAD_REQUEST);
+			return;
+		}
+		
+		JsonWebToken tokenSendBack = new JsonWebToken();
+		tokenSendBack.setOtherClaims("username", "admin");
+		
+		SecretKeySpec hmacSecretKeySpec = new SecretKeySpec(SmartSecretKeyHolder.getSecretKey(), JavaAlgorithm.HS256);
+		KeyWrapper keyWrapper = new KeyWrapper();
+		keyWrapper.setAlgorithm(Algorithm.HS256);
+		keyWrapper.setSecretKey(hmacSecretKeySpec);
+		SignatureSignerContext signer = new MacSignatureSignerContext(keyWrapper);
+		
+		String appToken = new JWSBuilder().jsonContent(tokenSendBack).sign(signer);
+		String encodedToken = URLEncoder.encode(appToken, StandardCharsets.UTF_8.name());
+		
+		String decoded = URLDecoder.decode(token, StandardCharsets.UTF_8.name());
+		response.sendRedirect(decoded.replace("{APP_TOKEN}", encodedToken));
```

Comment:
At this point, I think we should just return the `appToken` string and see what happens (the `appToken` should be a properly formatted and encoded JWT as a string... we might need to wrap it in an object, but it should ideally be something simple such as:

```json
{
    "token": "eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJzdWIiOiIxMjM0NTY3ODkwIiwibmFtZSI6IkpvaG4gRG9lIiwiYWRtaW4iOnRydWUsImp0aSI6IjUzYmFlOWMxLTQzMzQtNDBlZC1hYTUyLWM3ZGIyMWM4ODQyOCIsImlhdCI6MTYxMzc2MTM2NCwiZXhwIjoxNjEzNzY0OTY0fQ.FSSKvdvRBtUJ_jR3RT8T11JMNFS1jaX3goPZuJywzNs"
}
```

(Note that the non-sense inside the "token" field is really just what a signed JWT looks like).

Reply:
@ibacher we are passing the valid signed JWT token along with redirect url.

```
...&app-token=eyJhbGciOiJIUzI1NiJ9.eyJwYXRpZW50IjoiOWI3M2IxY2UtMjBmOS00YTM3LWE1N2ItNzkzMjc5MTg5ZjFmIn0.Bp8ESITgNhpd94F2snzrFcF23aLebbng-q1EQWSVEeo
```